### ISSUE TITLE
Element loses animation style on route animation

### ISSUE THREAD

**Issue by alucardu** [angular/angular#51472]
### Which @angular/* package(s) are the source of the bug?

animations, router

### Is this a regression?

No

### Description

I have a route animation between 2 components (A and B).  Component A has a element which can be toggled between collapsed or expanded which also uses a Angular animation. 

The issue is that when I collapse the element on component A and then navigate to component B the collapsed element becomes expanded, because it looses the styling (height: 40px) set by the angular animation on enter.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-ivy-vkfgng

### Please provide the environment you discovered this bug in (run `ng version`)

```true
"@angular/cli": "~14.0.0",
"@angular/core": "^16.2.1",
"typescript": "~4.7.2"
```

**Comment by selehadin-cyber** [comment]
the collapsing seems to work fine.
clicking on the "click" element routes to the red component and clicking on the picture expands or collapses the biography.

i am not sure where the issue is to be honest with you as there is nothing to collapse on the green component it just routes to the red component when clicking the link.

if there is anything i am missing please let me know

**Comment by alucardu** [comment]
Apologies for the confusion.  The unwanted behavior is when you navigate from red to green with a collapsed element. When the page transition the element will be fully visible.

**Comment by alucardu** [comment]
@selehadin-cyber I did some more investigation and I think I pin pointed the situation. 

The `other-component` has a animation defined `toggleContent` which allows for the `p` element to be expanded and collapsed.

When the `other-component` is destroyed (aka, when the route changes) it triggers the `void` state on the `toggleContent` animation. I think this is expected behavior but it's less than ideal for me.

**Comment by alucardu** [comment]
https://github.com/angular/angular/pull/50860/commits looks like this issue is being worked on.

**Comment by alucardu** [comment]
I've updated the stackbliz a little bit > https://stackblitz.com/edit/angular-ivy-vkfgng just click between the 2 links and you should see what I mean.

**Comment by alucardu** [comment]
> @alucardu Hi
> 
> yes you are right the void state does get triggered when changing routes. as a workaround you can use the following code for the void state and it gives the desired behavior
> 
> `state('void', style({ height: '40px' })),`

Except when the element is expanded. Then it will snap back to 40px from the expanded height. 

A fix for this is too put the element in another component and have the animation in there, not in the parent component. 

Or maybe I can pass the element client height to the animation and use that in the void height.

**Comment by alucardu** [comment]
I implemented a work-around: https://stackblitz.com/edit/angular-ivy-vkfgng I now set the height of the clicked element as a param to the animation so it uses that height on the void state.

**Comment by selehadin-cyber** [comment]
i think that is the way it should be done.
@JeanMeche could take a look at this to confirm. since i am not familiar with the animation side of the codebase

**Comment by JeanMeche** [comment]
Yes, this is definitely a workaround your issue. I would have done the same way.

**Comment by alucardu** [comment]
I improved it a little by abstracting the animation logic into a directive, pretty happy with the result.

**Comment by thePunderWoman** [comment]
This issue was closed as obsolete due to the introduction of the new `animate.enter` and `animate.leave` APIs. The `@angular/animations` package is now deprecated and will be removed in Angular v23. We hope you give the new APIs a try.
